pdp-server/src/api/horizon_fallback.rs

```diff
@@ -0,0 +1,272 @@
+use axum::{
+    body::{to_bytes, Body},
+    extract::State,
+    http::{Method, Request, Response, StatusCode},
+    response::IntoResponse,
+};
+use http::header::HeaderName;
+use pdp_engine::PDPError;
+use reqwest::header::HeaderValue;
+
+use crate::state::AppState;
+use pdp_engine::PDPEngine;
+
+/// Forward unmatched requests to the legacy Horizon PDP service (Python-based PDP)
+pub(super) async fn fallback_to_horizon(
+    State(state): State<AppState>,
+    req: Request<Body>,
+) -> impl IntoResponse {
+    // Get path for forwarding
+    let path = req.uri().path_and_query().map(|x| x.as_str()).unwrap_or("");
+
+    // Convert method to reqwest method
+    let method = match *req.method() {
+        Method::GET => Method::GET,
+        Method::POST => Method::POST,
+        Method::PUT => Method::PUT,
+        Method::DELETE => Method::DELETE,
+        Method::PATCH => Method::PATCH,
+        Method::HEAD => Method::HEAD,
+        Method::OPTIONS => Method::OPTIONS,
+        _ => {
```

Comment:
Why do you care if it's unsupported ? Let the server implement this, doing that would reduce the redundant Method::GET => Method::GET and same for all other methods